src/app/(views)/managed-courses/[courseId]/modules/[moduleId]/edit/page.tsx

```diff
@@ -112,6 +135,11 @@ export default function EditModule() {
       setValue('description', coursModule.description)
       setValue('type', coursModule.type)
       setValue('content', coursModule.content)
+
+      // Fetch the file from AWS if there's a content URL
+      if (coursModule.content) {
```

Comment:
Comment: "Fetch the file from AWS if there's a content URL"
```if (coursModule.content)``` does **not** check if content is a valid URL. It only checks if the value truthy. At the very least, also check that `courseModule.type != "text"`.